lib/BirthdayCalendarGenerator.php

```diff
@@ -0,0 +1,204 @@
+
+    }
+
+    /**
+     * Sets the input objects.
+     *
+     * You must either supply a vCard as a string or as a Component/VCard object.
+     * It's also possible to supply an array of strings or objects.
+     *
+     * @param mixed $objects
+     *
+     * @return void
+     */
+    function setObjects($objects) {
+
+        if (!is_array($objects)) {
+            $objects = [$objects];
+        }
+
+        $this->objects = [];
+        foreach ($objects as $object) {
+
+            if (is_string($object)) {
+
+                $vObj = Reader::read($object);
+                if (!$vObj instanceof Component\VCard) {
+                    throw new \InvalidArgumentException('String could not be parsed as \\Sabre\\VObject\\Component\\VCard by setObjects');
+                }
+
+                $this->objects[] = $vObj;
+
+            } elseif ($object instanceof Component\VCard) {
+
+                $this->objects[] = $object;
+
+            } else {
+
+                throw new \InvalidArgumentException('You can only pass strings or \\Sabre\\VObject\\Component\\VCard arguments to setObjects');
+
+            }
+
+        }
+
+    }
+
+    /**
+     * Sets the reference timezone for floating times.
+     *
+     * @param DateTimeZone $timeZone
+     *
+     * @return void
+     */
+    function setTimeZone(DateTimeZone $timeZone) {
+
+        $this->timeZone = $timeZone;
+
+    }
+
+    /**
+     * Sets the output format for the SUMMARY
+     *
+     * @param string $format
+     *
+     * @return void
+     */
+    function setFormat($format) {
+
+        $this->format = $format;
+
+    }
+
+    /**
+     * Parses the input data and returns a VCALENDAR.
+     *
+     * @return Component/VCalendar
+     */
+    function getResult() {
+
+        $calendar = new VCalendar();
+
+        foreach ($this->objects as $object) {
+
+            // Skip if there is no BDAY property.
+            if (!$object->select('BDAY')) {
+                continue;
+            }
+
+            // We're always converting to vCard 4.0 so we can rely on the
+            // VCardConverter handling the X-APPLE-OMIT-YEAR property for us.
+            $object = $object->convert(Document::VCARD40);
+
+            // Skip if the BDAY property is not of the right type.
+            if (!$object->BDAY instanceof Property\VCard\DateAndOrTime) {
+                continue;
+            }
+
+            // Skip if we can't parse the BDAY value.
+            try {
+                $dateParts = DateTimeParser::parseVCardDateTime($object->BDAY->getValue());
+            } catch (\InvalidArgumentException $e) {
+                continue;
+            }
+
+            // Set a year if it's not set.
+            if (!$dateParts['year']) {
+                $object->BDAY=$this->defaultYear . '-' . $dateParts['month'] . '-' . $dateParts['date'];
+
+                // @TODO:
+                // It may make sense to add an info to our event that the year was unknown.
```

Comment:
I agree, this does make sense.... perhaps we should add it as another value to X-SABRE-BIRTHDAY.